weave/trait/power/power_sources_trait.proto

```diff
@@ -187,5 +187,38 @@ message PowerSourcesTrait {
          *
          */
         repeated uint32                                      active_sources = 4;
+
+        // ----------- EVENTS ----------- //
+        message ActiveSourcesChangedEvent {
+                option (wdl.message_type) = EVENT;
+
+                option (wdl.event) = {
+                    id: 1,
+                    event_importance: EVENT_IMPORTANCE_PRODUCTION_STANDARD,
+                    reserved_tag_min: 1,
+                    reserved_tag_max: 2
+                };
+
+                /**
+                 *  This property represents as an array *all* of the _active_
+                 *  power sources in a device resource, sorted ascending on
+                 *  the power source order as the primary key and the power
+                 *  source status, filtering out Inactive status, as the
+                 *  secondary sort key. The array values are the instance
+                 *  identifiers of the respective power sources.
+                 *
+                 *  By using the instance identifiers of each power source as
+                 *  the array values, this allows publisher clients to easily
+                 *  and directly place the instance identifier into a
+                 *  subsequent publisher observation request to dive into
+                 *  additional details about a power source.
+                 *
+                 *  This representation also allows trait observers to easily
+                 *  dereference to the most important source, that in the
+                 *  first array slot, the primary, active source.
+                 *
+                 */
+                repeated uint32 active_sources = 1;
```

Comment:
Given a similar pattern we've used in other events, might be useful to put in a list of previously_active sources as well.

Reply:
Good idea